Can't compile Slic3r on Win 7 32bit

### Version

_slic3r-mswin-x86-1-2-9a-stable_
_strawberry-perl-5.22.2.1-32bit_
_boost_1_59_0_

![slic3r_compile_error](https://cloud.githubusercontent.com/assets/9814372/19543232/1520216e-9676-11e6-99cf-2f6a9bf6c592.gif)
### Operating system type + version

_Windows 7 Ultimate SP1_
### Behavior
- _Followed all instructions in "https://github.com/alexrj/Slic3r/wiki/Running-Slic3r-from-git-on-Windows". All ok until "perl Build.PL": compilation aborts when "building xs.dll file from 'src\perlglue.cpp' at line 129 of ....Windows.pm"
- _Steps needed to reproduce the problem_
  - _Just follow instructions at "https://github.com/alexrj/Slic3r/wiki/Running-Slic3r-from-git-on-Windows" without any change_
- _Actual Results_
  - _Screenshots enclosed_

_Is this a new feature request?_ NO

I browsed the forum and tried to force manual installation, tried many times, deleted and rebuilt all from scratch but the result didn't change.
Please help, I'm going nut!
Thanks.

I think I have the same issue: I was running slic3r and updateing it from time ti time useing GIT with no problems. Today I issued a git pull and then perl Buil.pl as always. This time I got this:

Generating main XS file...
gcc -c -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE  -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -fwrapv -fno-strict-aliasing -mms-bitfields -xc++ -Isrc/libslic3r -IC:\dev\boost_1_59_0 -D_GLIBCXX_USE_C99 -DHAS_BOOL -DNOGDI -DSLIC3RXS -DBOOST_ASIO_DISABLE_KQUEUE -DBOOST_LIBS -Isrc -Ibuildtmp -s -O2 -I"C:\STRAWB~1\perl\lib\CORE" -I"C:\STRAWB~1\c\include" -o "src\perlglue.o" "src\perlglue.cpp"
In file included from C:\dev\boost_1_59_0/boost/mpl/bind.hpp:50:0,
                 from C:\dev\boost_1_59_0/boost/mpl/lambda.hpp:18,
                 from C:\dev\boost_1_59_0/boost/mpl/apply.hpp:25,
                 from C:\dev\boost_1_59_0/boost/iterator/iterator_facade.hpp:37,
                 from C:\dev\boost_1_59_0/boost/range/iterator_range_core.hpp:27,
                 from C:\dev\boost_1_59_0/boost/range/iterator_range.hpp:13,
                 from C:\dev\boost_1_59_0/boost/range/as_literal.hpp:22,
                 from C:\dev\boost_1_59_0/boost/algorithm/string/trim.hpp:19,
                 from C:\dev\boost_1_59_0/boost/algorithm/string.hpp:19,
                 from src/libslic3r/Config.hpp:13,
                 from src/xsinit.h:48,
                 from src\perlglue.cpp:2:
C:\dev\boost_1_59_0/boost/mpl/aux_/include_preprocessed.hpp:37:90: fatal error: boost/mpl/aux_/preprocessed/gcc/PerlSock_bind.hpp: No such file or directory
 #   include BOOST_PP_STRINGIZE(boost/mpl/aux_/preprocessed/AUX778076_PREPROCESSED_HEADER)
                                                                                          ^
compilation terminated.
error building xs.dll file from 'src\perlglue.cpp' at C:/Strawberry/perl/lib/ExtUtils/CBuilder/Platform/Windows.pm line 129.
FAIL
! Installing ./xs failed. See C:\Users\Martin\.cpanm\work\1476981387.12604\build.log for details. Retry with --force to force install it.
The XS/C++ code failed to compile, aborting

--- a/xs/src/libslic3r/Config.cpp
+++ b/xs/src/libslic3r/Config.cpp
@@ -10,6 +10,8 @@
 #include <boost/foreach.hpp>
 #include <boost/property_tree/ini_parser.hpp>
 #include <boost/property_tree/ptree.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/lexical_cast.hpp>
 
 #if defined(_WIN32) && !defined(setenv) && defined(_putenv_s)
 #define setenv(k, v, o) _putenv_s(k, v)
@@ -405,5 +407,34 @@ StaticConfig::keys() const {
     }
     return keys;
 }
+
+bool
+ConfigOptionPoint::deserialize(std::string str, bool append) {
+    std::vector<std::string> tokens(2);
+    boost::split(tokens, str, boost::is_any_of(",x"));
+    try {
+        this->value.x = boost::lexical_cast<coordf_t>(tokens[0]);
+        this->value.y = boost::lexical_cast<coordf_t>(tokens[1]);
+    } catch (boost::bad_lexical_cast &e){
+        std::cout << "Exception caught : " << e.what() << std::endl;
+        return false;
+    }
+    return true;
+};
+
+bool
+ConfigOptionPoint3::deserialize(std::string str, bool append) {
+    std::vector<std::string> tokens(3);
+    boost::split(tokens, str, boost::is_any_of(",x"));
+    try {
+        this->value.x = boost::lexical_cast<coordf_t>(tokens[0]);
+        this->value.y = boost::lexical_cast<coordf_t>(tokens[1]);
+        this->value.z = boost::lexical_cast<coordf_t>(tokens[2]);
+    } catch (boost::bad_lexical_cast &e){
+        std::cout << "Exception caught : " << e.what() << std::endl;
+        return false;
+    }
+    return true;
+};
 
 }

--- a/xs/src/libslic3r/Config.hpp
+++ b/xs/src/libslic3r/Config.hpp
@@ -10,8 +10,6 @@
 #include <stdexcept>
 #include <string>
 #include <vector>
-#include <boost/algorithm/string.hpp>
-#include <boost/lexical_cast.hpp>
 #include "libslic3r.h"
 #include "Point.hpp"
 

--- a/xs/src/libslic3r/Config.hpp
+++ b/xs/src/libslic3r/Config.hpp
@@ -353,18 +353,7 @@ class ConfigOptionPoint : public ConfigOptionSingle<Pointf>
         return ss.str();
     };
     
-    bool deserialize(std::string str, bool append = false) {
-        std::vector<std::string> tokens(2);
-        boost::split(tokens, str, boost::is_any_of(",x"));
-        try {
-            this->value.x = boost::lexical_cast<coordf_t>(tokens[0]);
-            this->value.y = boost::lexical_cast<coordf_t>(tokens[1]);
-        } catch (boost::bad_lexical_cast &e){
-            std::cout << "Exception caught : " << e.what() << std::endl;
-            return false;
-        }
-        return true;
-    };
+    bool deserialize(std::string str, bool append = false);
 };
 
 class ConfigOptionPoint3 : public ConfigOptionSingle<Pointf3>
@@ -384,19 +373,7 @@ class ConfigOptionPoint3 : public ConfigOptionSingle<Pointf3>
         return ss.str();
     };
     
-    bool deserialize(std::string str, bool append = false) {
-        std::vector<std::string> tokens(3);
-        boost::split(tokens, str, boost::is_any_of(",x"));
-        try {
-            this->value.x = boost::lexical_cast<coordf_t>(tokens[0]);
-            this->value.y = boost::lexical_cast<coordf_t>(tokens[1]);
-            this->value.z = boost::lexical_cast<coordf_t>(tokens[2]);
-        } catch (boost::bad_lexical_cast &e){
-            std::cout << "Exception caught : " << e.what() << std::endl;
-            return false;
-        }
-        return true;
-    };
+    bool deserialize(std::string str, bool append = false);
     
     bool is_positive_volume () {
         return this->value.x > 0 && this->value.y > 0 && this->value.z > 0;

